plugins/java-decompiler/engine/src/org/jetbrains/java/decompiler/struct/ContextUnit.java

```diff
@@ -128,6 +131,9 @@ public void save() {
         // classes
         for (int i = 0; i < classes.size(); i++) {
           StructClass cl = classes.get(i);
+          if (!cl.isOwn()) {
+            continue;
+          }
```

Comment:
This second change is not needed. The problem demonstrated by your scenario is that all plain .class files go into the default context unit named '""', created in `StructContext`'s constructor with `own=true`. Context units with types other than `TYPE_FOLDER` shouldn't be affected.

Reply:
Ah yes, probably not needed. I think I added this defensively i.e. there may be a mix of own and not-own classes in the future. I guess it's fine to remove this.